File: vsc/src/smelly/DeepNestingExample.js
/**
 * Deep Nesting - Too many nested control structures.
 * CodeScene flags methods with excessive indentation levels.
 */

export class DeepNestingExample {
  validateAndProcessUser(
    username,
    password,
    email,
    age,
    country,
    hasAgreedToTerms
  ) {
    if (username != null) {
      if (username.length >= 3) {
        if (username.length <= 20) {
          if (password != null) {
            if (password.length >= 8) {
              if (/[A-Z]/.test(password)) {
                if (/\d/.test(password)) {
                  if (email != null) {
                    if (email.includes("@")) {
                      if (email.includes(".")) {
                        if (age >= 13) {
                          if (age <= 120) {
                            if (country != null) {
                              if (["US", "UK", "CA"].includes(country)) {
                                if (hasAgreedToTerms) {
                                  console.log("User validation successful");
                                  this.createUserAccount(
                                    username,
                                    password,
                                    email
                                  );
                                } else {
                                  console.error("User must agree to terms");
                                }
                              } else {
                                console.error(
                                  "Service not available in country"
                                );
                              }
                            } else {
                              console.error("Country is required");
                            }
                          } else {
                            console.error("Age must be realistic");
                          }
                        } else {
                          console.error("User must be at least 13 years old");
                        }
                      } else {
                        console.error("Email must contain domain extension");
                      }
                    } else {
                      console.error("Email must contain @ symbol");
                    }
                  } else {
                    console.error("Email is required");
                  }
                } else {
                  console.error("Password must contain at least one number");
                }
              } else {
                console.error(
                  "Password must contain at least one uppercase letter"
                );
              }
            } else {
              console.error("Password must be at least 8 characters");
            }
          } else {
            console.error("Password is required");
          }
        } else {
          console.error("Username too long");
        }
      } else {
        console.error("Username too short");
      }
    } else {
      console.error("Username is required");
    }
  }

  createUserAccount(username, password, email) {
    console.log(`Creating account for: ${username}`);
  }
}

File: jetbrains/src/main/java/SmellyTest/DeepNestingExample.java
package SmellyTest;

/**
 * Deep Nesting Example - Too many nested control structures
 * CodeScene flags methods with excessive indentation levels
 */
public class DeepNestingExample {

    public void validateAndProcessUser(String username, String password, String email,
                                       int age, String country, boolean hasAgreedToTerms) {
        if (username != null) {
            if (username.length() >= 3) {
                if (username.length() <= 20) {
                    if (password != null) {
                        if (password.length() >= 8) {
                            if (password.matches(".*[A-Z].*")) {
                                if (password.matches(".*[0-9].*")) {
                                    if (email != null) {
                                        if (email.contains("@")) {
                                            if (email.contains(".")) {
                                                if (age >= 13) {
                                                    if (age <= 120) {
                                                        if (country != null) {
                                                            if (country.equals("US") || country.equals("UK") || country.equals("CA")) {
                                                                if (hasAgreedToTerms) {
                                                                    System.out.println("User validation successful");
                                                                    // Actually process the user
                                                                    createUserAccount(username, password, email);
                                                                } else {
                                                                    System.err.println("User must agree to terms");
                                                                }
                                                            } else {
                                                                System.err.println("Service not available in country");
                                                            }
                                                        } else {
                                                            System.err.println("Country is required");
                                                        }
                                                    } else {
                                                        System.err.println("Age must be realistic");
                                                    }
                                                } else {
                                                    System.err.println("User must be at least 13 years old");
                                                }
                                            } else {
                                                System.err.println("Email must contain domain extension");
                                            }
                                        } else {
                                            System.err.println("Email must contain @ symbol");
                                        }
                                    } else {
                                        System.err.println("Email is required");
                                    }
                                } else {
                                    System.err.println("Password must contain at least one number");
                                }
                            } else {
                                System.err.println("Password must contain at least one uppercase letter");
                            }
                        } else {
                            System.err.println("Password must be at least 8 characters");
                        }
                    } else {
                        System.err.println("Password is required");
                    }
                } else {
                    System.err.println("Username too long");
                }
            } else {
                System.err.println("Username too short");
            }
        } else {
            System.err.println("Username is required");
        }
    }

    private void createUserAccount(String username, String password, String email) {
        System.out.println("Creating account for: " + username);
    }
}
